port/unix/omrsignal.c

```diff
@@ -1193,11 +1259,8 @@ registerSignalHandlerWithOS(OMRPortLibrary *portLibrary, uint32_t portLibrarySig
 		}
 	}
 
-	/* CMVC 96193 signalsWithMasterHandlers is checked without acquiring the masterHandlerMonitor */
-	issueWriteBarrier();
```

Comment:
According to the comment, the problem is not that registerSignalHandlerWithOS is called without acquiring registerHandlerMonitor, the problem is that signalsWithHandlers is used elsewhere without first acquiring registerHandlerMonitor, and all the actions previous to the write barrier need to be finished before the bit set.

Reply:
CMVC 96193 contains the following reasoning. The idea was to avoid unnecessary acquires and improve performance, change sig_protect to improve perf. 

"We have seen j9thread_monitor_entry and exit for the masterHandlerMonitor mutex contributing 914
instructions twice in the newDirectByteBuffer. 

High-level logic is
acquire masterHandlerMonitor mutex
check for -Xrs flag
if not set then call registerMasterHandlers
release masterHandlerMonitor mutex

If we can test the -Xrs flag first and only acquire/release the mutex if we are going to call 
registerMasterHandlers then we can strip 1828 instructions from the newDirectByteBuffer pathlength.
"

Reply:
will undo this change.

Reply:
the following commit: `Remove unnecessary write barrier`, has been removed from the pull request.

Reply:
hmm, I don't think it needed to be removed. I was commenting because you seemed to misunderstand the reason for the write barrier, and providing the context for the barrier. Looking at the sig_protect code, it doesn't use signalsWithHandlers to check the -Xrs flag, but signalOptionsGlobal. I assume the code was changed at some point. Perhaps there is opportunity for future optimization of the monitors, but this should be left out of this change.

Reply:
`signalsWithHandlers` and `signalsWithMasterHandlers` are always accessed with `registerHandlerMonitor`. there is a comment about this next to their description: `Access to this must be protected by the registerHandlerMonitor`. should i re-add the commit for removing the write barrier?

Reply:
Re-adding it is fine with me.